Generate RPC metrics from YAML

joaopgrassi opened the pull request:
Address part of #74 

CC @jsuereth @trask @lmolkova @jack-berg as previous authors

joaopgrassi commented on semantic_conventions/metrics/rpc-metrics.yaml:
```diff
@@ -0,0 +1,116 @@
+groups:
+  # TODO: Should we list the attributes on each metric
```
All: The YAML now allow us to define the attributes under each metric. That will render the metric table and its attributes. 

Given here all attributes apply to all metrics, should we:

- Be consistent and add all attributes to all metrics. This will duplicate the same attribute tables for each metric. 
- Leave the attributes in its own section like we have today. This reduces duplication a lot with the "downside" of not having the attributes close to the metrics in the page. But given there aren't many metrics, I think it's still OK.

jsuereth replied:
I actually think we should do the first BUT for this PR we can keep status quo with the second.  I would NOT close the underling ticket though.

joaopgrassi commented on semantic_conventions/metrics/rpc-metrics.yaml:
```diff
@@ -0,0 +1,116 @@
+    metric_name: rpc.client.response.size
+    brief: >
+      Measures the size of RPC response messages (uncompressed).
+      **Streaming**: Recorded per response in a streaming batch
```
We had a column for "Streaming" which we don't now. The workaround I used is to add it in the description. Do you think that's enough? I also thought of adding it as a Note, but I find that even more "hidden".

joaopgrassi replied:
I added the "streaming" text for `rpc.server.duration` and `rpc.client.duration` as a Note. Now that we target the build-tools 0.19, notes are also rendered in markdown. Closing this as I think it's good enough.

joaopgrassi commented on semantic_conventions/metrics/rpc-metrics.yaml:
```diff
@@ -0,0 +1,116 @@
+  # OR leave a single table like it is today? Since all attributes are applied
+  # to all metrics, the repetition of them bloats the page
+  - id: attributes.metrics.rpc
+    type: attribute_group
```
If we are OK in leaving the attributes in their own table, I will then remove this `attribute_group` here.

jsuereth replied:
I think we should probably have the attributes per-metric here. I *think* client vs. server attributes are actually different, I'm not sure where the language around that got confused.

joaopgrassi commented:
I will work on solving the conflicts and make sure the issue isn't closed my this PR.

joaopgrassi commented:
@jsuereth I updated the PR, solved the conflicts. This should be good to go. 

To address your comments, I updated the issue #74 adding the left-over to generate the attributes for all metrics as a follow up. The ticket won't be closed as I changed the description in the PR (I hope!)

For the other comment about re-examine the wording/attributes, I created another issue #219

jsuereth reviewed (approved):
Thanks for fixing this up!  LGTM for the incremental progress.
